src/libcore/str/mod.rs

```diff
@@ -1484,19 +1485,452 @@ impl<'a> DoubleEndedIterator for LinesAny<'a> {
 #[allow(deprecated)]
 impl FusedIterator for LinesAny<'_> {}
 
+#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+mod avx2 {
```

Comment:
Shouldn't this specify a bunch of target features? If you don't do so it generates MUCH worse code (and maybe it doesn't even work, not sure).

Reply:
@CryZe good catch, I updated it the guard to use `target_feature = "avx2"` that should solve this